internal/util/importutilv2/parquet/util.go

```diff
@@ -102,77 +93,116 @@ func CreateFieldReaders(ctx context.Context, fileReader *pqarrow.FileReader, sch
+		return &arrow.Int16Type{}
 	case schemapb.DataType_Int32:
-		return typeutil.IsArithmetic(dst) && dst != schemapb.DataType_Int8 && dst != schemapb.DataType_Int16
+		return &arrow.Int32Type{}
 	case schemapb.DataType_Int64:
-		return typeutil.IsFloatingType(dst) || dst == schemapb.DataType_Int64
+		return &arrow.Int64Type{}
 	case schemapb.DataType_Float:
-		if isList && dst == schemapb.DataType_FloatVector {
-			return true
-		}
-		return typeutil.IsFloatingType(dst)
+		return &arrow.Float32Type{}
 	case schemapb.DataType_Double:
-		if isList && dst == schemapb.DataType_FloatVector {
-			return true
-		}
-		return dst == schemapb.DataType_Double
-	case schemapb.DataType_String, schemapb.DataType_VarChar:
-		return typeutil.IsStringType(dst) || typeutil.IsJSONType(dst)
+		return &arrow.Float64Type{}
+	case schemapb.DataType_VarChar, schemapb.DataType_String:
+		return &arrow.StringType{}
+	case schemapb.DataType_Array:
+		return &arrow.ListType{}
 	case schemapb.DataType_JSON:
-		return typeutil.IsJSONType(dst)
+		return &arrow.StringType{}
 	case schemapb.DataType_BinaryVector:
-		return dst == schemapb.DataType_BinaryVector
-	case schemapb.DataType_Float16Vector:
-		return dst == schemapb.DataType_Float16Vector
+		return arrow.ListOfField(arrow.Field{
+			Name:     "item",
+			Type:     &arrow.Uint8Type{},
+			Nullable: true,
+			Metadata: arrow.Metadata{},
+		})
+	case schemapb.DataType_FloatVector:
+		return arrow.ListOfField(arrow.Field{
+			Name:     "item",
+			Type:     &arrow.Float32Type{},
+			Nullable: true,
+			Metadata: arrow.Metadata{},
+		})
+	case schemapb.DataType_Float16Vector, schemapb.DataType_BFloat16Vector:
+		return arrow.ListOfField(arrow.Field{
+			Name:     "item",
+			Type:     &arrow.Uint8Type{},
+			Nullable: true,
+			Metadata: arrow.Metadata{},
+		})
 	default:
-		return false
+		panic("unsupported data type")
+	}
+}
+
+func isAutoPKField(field *schemapb.FieldSchema) bool {
+	return field.GetIsPrimaryKey() && field.GetAutoID()
+}
+
+func ConvertToArrowSchema(schema *schemapb.CollectionSchema) *arrow.Schema {
+	fields := make([]arrow.Field, 0)
+	for _, field := range schema.GetFields() {
+		if isAutoPKField(field) {
+			continue
+		}
+		if field.GetDataType() == schemapb.DataType_Array {
+			fields = append(fields, arrow.Field{
+				Name: field.GetName(),
+				Type: arrow.ListOfField(arrow.Field{
+					Name:     "item",
+					Type:     convertToArrowType(field.GetElementType()),
+					Nullable: true,
+					Metadata: arrow.Metadata{},
+				}),
+				Nullable: true,
+				Metadata: arrow.Metadata{},
+			})
+			continue
+		}
+		fields = append(fields, arrow.Field{
+			Name:     field.GetName(),
+			Type:     convertToArrowType(field.GetDataType()),
+			Nullable: true,
+			Metadata: arrow.Metadata{},
+		})
+	}
+	return arrow.NewSchema(fields, nil)
+}
+
+func isSchemaEqual(schema *schemapb.CollectionSchema, arrSchema *arrow.Schema) error {
+	i := 0
+	for _, field := range schema.GetFields() {
```

Comment:
just remove i, and `for i, field := range schema.GetFields() {`

Reply:
if isAutoPKField() is true, i should not increase